app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt

```diff
@@ -300,8 +300,16 @@ public open class MainPresenterImpl(val mapzenLocation: MapzenLocation, val bus:
         }
     }
 
+    override fun onExitNavigation() {
+        vsm.viewState = ViewStateManager.ViewState.SEARCH_RESULTS
+        routingEnabled = false;
+        routeManager?.reverse = false
+        onFindMeButtonClick()
+    }
+
     override fun onMapMotionEvent(): Boolean {
         mainViewController?.rotateCompass()
+        onFindMeButtonClick()
```

Comment:
What's the intent of this call? The effect I see is that whenever I rotate the map the position is snapped back to my current location.